server/auth/github.js

```diff
@@ -0,0 +1,199 @@
+// @flow
+import passport from "@outlinewiki/koa-passport";
+import Router from "koa-router";
+import { Strategy as GithubStrategy } from "passport-github2";
+import accountProvisioner from "../commands/accountProvisioner";
+import env from "../env";
+import * as Github from "../github";
+import auth from "../middlewares/authentication";
+import passportMiddleware from "../middlewares/passport";
+import { Authentication, Collection, Integration, Team } from "../models";
+import { StateStore } from "../utils/passport";
+
+const router = new Router();
+const providerName = "github";
+const GITHUB_CLIENT_ID = process.env.GITHUB_KEY;
+const GITHUB_CLIENT_SECRET = process.env.GITHUB_SECRET;
+
+const scopes = ["repo", "read:org", "user"];
+
+export const config = {
+  name: "Github",
+  enabled: !!GITHUB_CLIENT_ID,
+};
+
+if (GITHUB_CLIENT_ID) {
+  const strategy = new GithubStrategy(
+    {
+      clientID: GITHUB_CLIENT_ID,
+      clientSecret: GITHUB_CLIENT_SECRET,
+      callbackURL: `${env.URL}/auth/github.callback`,
+      passReqToCallback: true,
+      store: new StateStore(),
+      scope: scopes,
+    },
+    async function (req, accessToken, refreshToken, profile, done) {
+      try {
+        const result = await accountProvisioner({
+          ip: req.ip,
+          team: {
+            name: profile.displayName,
+            avatarUrl: profile._json.avatar_url,
+          },
+          user: {
+            name: profile.username,
+            email: profile.emails[0].value,
+            avatarUrl: profile._json.avatar_url,
+          },
+          authenticationProvider: {
+            name: providerName,
+            providerId: profile.id,
+          },
+          authentication: {
+            providerId: profile.id,
+            accessToken,
+            refreshToken,
+            scopes,
+          },
+        });
+        return done(null, result.user, result);
+      } catch (err) {
+        return done(err, null);
+      }
+    }
+  );
+
+  // For some reason the author made the strategy name capatilised, I don't know
+  // why but we need everything lowercase so we just monkey-patch it here.
+  strategy.name = providerName;
+  passport.use(strategy);
+
+  router.get(
+    "github",
+    passport.authenticate(providerName, { scope: scopes, session: false })
+  );
+
+  // start the oauth process and redirect user to Github
+  // signin callback from Github
+  router.get(
+    "github.callback",
+    auth({ required: false }),
+    passportMiddleware(providerName)
+  );
+
+  router.get("github.commands", auth({ required: false }), async (ctx) => {
```

Comment:
This method and `github.post` don't seem to make sense, they're only applicable to the Slack integration